botocore/compat.py

```diff
@@ -349,3 +349,10 @@ def get_tzinfo_options():
     from collections.abc import MutableMapping
 except ImportError:
     from collections import MutableMapping
+
+# Detect if CRT is available for use
+try:
+    import awscrt.auth
+    HAS_CRT = True
```

Comment:
Yep, completely agree on the user-agent addition. I think we can get that added. For the environment variable opt out, I think that's reasonable to give the option for a quick fix. You could similarly uninstall the crt from the workspace, but I guess there may be a future where the Python CRT is used across multiple tools, not just boto3.

We don't really have access to the providers here, so we could do a one off grab from os.environ inside of this try/except closure, otherwise I'll likely need to add a function somewhere else that we use instead of `HAS_CRT` to evaluate all required criteria for CRT use.